pkg/cleanup/cleanup.go

```diff
@@ -15,9 +16,9 @@ func GetTagsMatchingPrefixes(prefixes, tags *[]string) []string {
 	if len(*prefixes) > 0 && len(*tags) > 0 {
 		for _, prefix := range *prefixes {
 			for _, tag := range *tags {
-				if strings.HasPrefix(tag, prefix) {
+				if match(tag, prefix, commitTag) {
 					matchingTags = append(matchingTags, tag)
-					log.Debugf("GetTagsMatchingPrefixes | Tag %s has prefix %s", tag, prefix)
+					log.Debugf("GetTagsMatchingPrefixes | Tag %s matched with %s", tag, prefix)
```

Comment:
1) Please never copy-paste method names into the debug log message, they will not be affected/recognized by any refactoring methods (e.g. renaming of the method) of any IDEs. The logging framework supports a config flag where the exact file name and line number will be printed too if desired, no need for manual shenanigans :)
2) Use logging fields rather than string format. That makes the logs searchable with a dedicated field should it be logged by any ELK/EFK instances. e.g. `log.WithFields(&log.Fields{"tag": tag, "prefix": prefix}).Debug("Tag matched")`

Reply:
Updated other logs as well to reflect what has been proposed